### PR TITLE
Aurora pipeline transform and tests finished: Caleb

### LINKED ISSUE #33: Create Hourly Aurorawatch transform.py
As an amateur stargazer, I want the latest aurora information from aurorawatch API to be cleaned and filtered, so I can determine if it is worth 'auroragazing'.

- Use base python for this

### LINKED ISSUE #39: Test Hourly Aurorawatch transform.py
As a developer, I want to test the Transform.py script to verify that the raw aurora data is correctly processed, cleaned, and transformed, so I can ensure that the data is ready for storage and future analysis without errors or inconsistencies.

- Use `pytest` for this

### PR DESCRIPTION
**Pull request by CalebMathers**
This closes #33 
This closes #39 

Wrote the tests for the transform script for AuroraWatch pipeline, then wrote functions to pass the tests.

### CODE REVIEW

File: aurorawatch-pipeline/test_aurora_transform.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+import pytest
+
+from aurora_transform import get_colour_id, get_current_datetime, transform
+
+
+class TestGetColourID():
+    '''Tests for the get colour id function.'''
+
+    @pytest.mark.parametrize('colour', [('green'), ('   green  '),
```
Comment (MurasatoNaoya) [inline aurorawatch-pipeline/test_aurora_transform.py:15]: Good use of parameterisation for testing Aurora colour indicators in a variety of weirder formats.

File: aurorawatch-pipeline/test_aurora_transform.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+from unittest.mock import patch
+from datetime import datetime as dt
+
+import pytest
+
+from aurora_transform import get_colour_id, get_current_datetime, transform
+
+
+class TestGetColourID():
```
Comment (MurasatoNaoya) [inline aurorawatch-pipeline/test_aurora_transform.py:12]: I like the choice of using test classes, tests are clearly groped this way making your code more readable, and they can be run as a group in terminal a lot easier too.

File: aurorawatch-pipeline/aurora_transform.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        raise TypeError('The colour must be given in string format.')
+
+    colour_id = ALERT_COLOURS.get(colour.strip().title())
+
+    if colour_id:
+        logging.info('Colour ID found.')
+        return colour_id
+
+    err_msg = f'No colour ID found for colour: {colour}'
+    logging.error(err_msg)
```
Comment (MurasatoNaoya) [inline aurorawatch-pipeline/aurora_transform.py:25]: Good use of logging here, so of our other tests were a little light on logging, so it's a important inclusion here. Especially as the primary thing we are fetching from this API is the colour indicator, it's a big deal if we can't fetch it for whatever reason.

**Review by MurasatoNaoya — APPROVED** [review]
Solid transform script and tests. LGTM. Let's keep the logging up for the remainder of our tests.

File: aurorawatch-pipeline/aurora_transform.py (hunk 3 above)
Comment (AlexOBarnes) [inline aurorawatch-pipeline/aurora_transform.py:25]: Good error handling

File: aurorawatch-pipeline/test_aurora_transform.py (hunk 1 above)
Comment (AlexOBarnes) [inline aurorawatch-pipeline/test_aurora_transform.py:15]: Really like the parameterised tests

File: aurorawatch-pipeline/test_aurora_transform.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    def test_get_colour_id_invalid_colour(self, colour):
+        '''Tests that if a colour is not found in the available alert
+        colours, an error is raised.'''
+        with pytest.raises(KeyError) as err:
+            get_colour_id(colour)
+
+        assert err.value.args[0] == f'No colour ID found for colour: {colour}'
+
+
+class TestGetCurrentDatetime():
```
Comment (AlexOBarnes) [inline aurorawatch-pipeline/test_aurora_transform.py:43]: Really comprehensive tests

**Review by AlexOBarnes — APPROVED** [review]
Looks good to me

**Pull request merged**